lgc/patch/PatchBufferOp.cpp

```diff
@@ -923,7 +934,8 @@ void PatchBufferOp::postVisitMemCpyInst(MemCpyInst &memCpyInst) {
     while (stride != 1) {
       // We only care about dword alignment (4 bytes) so clamp the max check here to that.
       const unsigned minStride = std::min(stride, 4u);
-      if (destAlignment.valueOrOne() >= minStride && srcAlignment.valueOrOne() >= minStride && (constantLength % stride) == 0)
+      if (destAlignment.valueOrOne() >= minStride && srcAlignment.valueOrOne() >= minStride &&
```

Comment:
Spurious formatting change. I know the formatting is wrong, but we should not be correcting formatting in code not otherwise touched by a change, unless the change is specifically to correct formatting.

Reply:
Ok. I got the rule now.

> High-level comment:
> 
> Why is this needed at all? If a null descriptor has all bits 0, extracting the length from it would give 0 anyway.

Yes. The extracted value gives 0. But the final array length needs another one or two operations base the extracted value. If it has an address offset, there is a sub operation then divide the stride to obtain the array length.

Reply:
Ah, I see. In SPIR-V, a variable-length array is always the last thing in the buffer, and there might be other fixed-length things before it. Maybe we need to add some code to transValueWithOpcode<OpArrayLength> in SPIRVReader.cpp. The problem is that you cannot see the pipeline options at that point. I'll have a think...

Reply:
How about this as a solution:

* In Builder::CreateGetBufferDescLength, add a new second arg, which is an integer value you want to be subtracted from the desc length. Document that the result is clamped at 0 if robustAccess is on.
* In transValueWithOpcode<OpArrayLength>(), pass the byte offset of the variable array as that new second arg.
* In DescBuilder, change the lgc.late.buffer.desc.length to pass that second arg.
* In PatchBufferOp, generate a subtract of that value from the length. If robustAccess is on, use a select to clamp the result at 0.

Reply:
Good idead. Thanks for your clear guideline.